src/main/java/org/openrewrite/java/migrate/MethodExceptionReplacerRecipe.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.ChangeType;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.FindMethods;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Map;
+
+public class MethodExceptionReplacerRecipe extends Recipe {
+    private final Map<String, Map<String, String>> methodToExceptionMapping;
+
+    public MethodExceptionReplacerRecipe(Map<String, Map<String, String>> methodToExceptionMapping) {
+        this.methodToExceptionMapping = methodToExceptionMapping;
+    }
```

Comment:
We don't support Map options; you can see this fail in the validation; can this be revised to individual `@Option` fields? Or a dedicated recipe as opposed to a generic one?
```
        Missing{property='MethodExceptionReplacerRecipe.methodToExceptionMapping', value='null', message='is required'}],
        Missing{property='MethodExceptionReplacerRecipe.methodToExceptionMapping', value='null', message='is required'}]
```

Reply:
Created a dedicated Recipe.